packages/embark-ui/src/components/ContractsDeployment.js

```diff
@@ -127,13 +128,31 @@ class Web3Contract extends React.Component {
     return this.inputsAsArray().length !== constructor.inputs.length;
   }
 
+  isDeployed() {
+    const contractInStore = this.props.web3ContractsDeployed[this.props.contract.className];
+    return contractInStore && contractInStore.isDeployed;
+  }
+  isCheckingIfDeployed() {
+    const contractInStore = this.props.web3ContractsDeployed[this.props.contract.className];
+    return contractInStore && contractInStore.running;
+  }
+
   render() {
     const abiConstructor = findConstructor(this.props.contract.abiDefinition);
     const argumentsRequired = abiConstructor && abiConstructor.inputs.length > 0;
+    const isDeployed = this.isDeployed();
+    const isCheckingIfDeployed = this.isCheckingIfDeployed();
     return (
-      <LayoutContract contract={this.props.contract} cardTitle={this.props.contract.className}>
+      <LayoutContract contract={this.props.contract} cardTitle={
+        <React.Fragment>
+          {this.props.contract.className}&nbsp;
+          {isCheckingIfDeployed && <FontAwesome name="spinner" spin />}
+          {!isCheckingIfDeployed && <span>{(isDeployed && `deployed at ${this.props.contract.deployedAddress}`) || 'not deployed'}</span>}
+        </React.Fragment>
```

Comment:
Putting jsx inside a components prop is a bit weird. A better way to od it would be to just pass `isCheckingIfDeployed ` and `isDeployed` to the component and let it render its title as need.

Unless this component is used somewhere else and it's a completely different way to render the title

Reply:
Yeah, I definitely agree, it was kind of how things were done for titles throughout Cockpit, but I've updated this one :) Definitely looks cleaner :)